src/sage/plot/misc.py

```diff
@@ -139,6 +139,13 @@ def setup_for_eval_on_grid(funcs,
     else:
         vars, free_vars = unify_arguments(funcs)
 
+    #check for invalid range entered by user
```

Comment:
needs a space `# check` I would write `# check for valid range`

Reply:
> Look into the Build & Test check. There are lots of failed doctests. It seems that `ranges[1]` may not exist.

Solved this. Some functions take range input differently, so nested my if condition inside another which checks if:
```
len(ranges) > 1
```
Haven't pushed it yet due to the issue mentioned in the comment below.